### PR TITLE
Update appui abstract tests

### PR DESCRIPTION
**Pull request by calebmshafer**
Updating the appui-abstract testing setup to no longer need a custom mocha config and to take advantage of Certa similar to core-frontend.

Slimmed down the dev dependency list in appui-abstract and re-factored tests as necessary.

### CODE REVIEW

**Review by NancyMcCallB — APPROVED** [review]

File: ui/appui-abstract/src/test/utils/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          test: /azure-storage|AzureFileHandler|UrlFileHandler/,
+          use: "null-loader"
+        },
+      ]
+    },
+    stats: "errors-only",
+    optimization: {
+      nodeEnv: "production"
+    },
+    externals: {
```
[the remark this answers is not in the document]
Comment (calebmshafer) [reply ui/appui-abstract/src/test/utils/webpack.config.js:46]: yup, copy and paste job. Fixed.

File: ui/appui-abstract/src/test/utils/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        /draco_encoder_nodejs.js$/
+      ],
+      rules: [
+        {
+          test: /\.js$/,
+          use: "source-map-loader",
+          enforce: "pre"
+        },
+        {
+          test: /azure-storage|AzureFileHandler|UrlFileHandler/,
```
[the remark this answers is not in the document]
Comment (calebmshafer) [reply ui/appui-abstract/src/test/utils/webpack.config.js:37]: fixed

File: ui/appui-abstract/src/test/utils/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    mode: "development",
+    entry: glob.sync(path.resolve(frontendLib, "test/**/*.test.js")),
+    output: {
+      path: path.resolve(frontendLib, "test/webpack/"),
+      filename: "bundled-tests.js",
+      devtoolModuleFilenameTemplate: "file:///[absolute-resource-path]"
+    },
+    devtool: "nosources-source-map",
+    module: {
+      noParse: [
```
[the remark this answers is not in the document]
Comment (calebmshafer) [reply ui/appui-abstract/src/test/utils/webpack.config.js:23]: fixed

File: ui/appui-abstract/src/test/properties/PropertyRecord.test.ts [hunk 4]
```diff
@@ -64,20 +64,15 @@ describe("PropertyRecord", () => {
 
   describe("fromString", () => {
 
-    it("should create a valid PropertyRecord with provided description", () => {
-      const description = getPropertyDescription();
-      const record = PropertyRecord.fromString("test", description);
-      expect(record).to.matchSnapshot();
```
[the remark this answers is not in the document]
Comment (calebmshafer) [reply ui/appui-abstract/src/test/properties/PropertyRecord.test.ts:70]: removed

File: ui/appui-abstract/src/test/utils/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      ]
+    },
+    stats: "errors-only",
+    optimization: {
+      nodeEnv: "production"
+    },
+    externals: {
+      electron: "commonjs electron",
+    },
+    plugins: [
```
[the remark this answers is not in the document]
Comment (calebmshafer) [reply ui/appui-abstract/src/test/utils/webpack.config.js:35]: It doesn't technically grab anything out of the process but if I don't set it sinon complains so I left it in there for now.

**Review by MarcNeely — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Pull request merged**